src/sentry/integrations/github/client.py

```diff
@@ -104,73 +104,74 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> JSONData:
 
         return tree
 
-    def get_repo_files(
-        self, repo_full_name: str, tree_sha: str, only_source_code_files: bool = True
+    def get_cached_repo_files(
+        self,
+        repo_full_name: str,
+        tree_sha: str,
+        only_source_code_files: bool = True,
+        cache_seconds: int = 3600 * 24,
     ) -> List[str]:
         """It return all files for a repo or just source code files.
 
         repo_full_name: e.g. getsentry/sentry
         tree_sha: A branch or a commit sha
         only_source_code_files: Include all files or just the source code files
         """
-        repo_files = []
-        try:
-            tree = self.get_tree(repo_full_name, tree_sha)
-            if tree is not None:
-                repo_files = [x["path"] for x in tree if x["type"] == "blob"]
-                if only_source_code_files:
-                    repo_files = filter_source_code_files(files=repo_files)
-        except Exception:
-            logger.exception("An unknown error has ocurred.")
+        key = f"github:repo:{repo_full_name}:{'source-code' if only_source_code_files else 'all'}"
+        repo_files: List[str] = cache.get(key, [])
+        if not repo_files:
+            try:
+                tree = self.get_tree(repo_full_name, tree_sha)
+                if tree is not None:
+                    repo_files = [x["path"] for x in tree if x["type"] == "blob"]
+                    if only_source_code_files:
+                        repo_files = filter_source_code_files(files=repo_files)
+                    # The backend's caching will skip silently if the object size greater than 5MB
+                    # The trees API does not return structures larger than 7MB
```

Comment:
What does it do for very large trees?

Reply:
@markstory GH will return a partial tree with `truncated` set to `True`. We can still do something useful based on a partial tree.

This barely happens. The repo needs to be really big. Ideally we will fix it. It's not too difficult.